Population/Population Datamart/Update Datamart Small Area Population Estimates.R

```diff
@@ -0,0 +1,135 @@
+### 1 - Information ----
+
+# Codename - Update Datamart Small Area Population Estimates
+# Original Author - Calum Purdie
+# Original Date - 31/10/2019
+# Updated - 04/11/2019
+# Updated - 
+# Type - Updating files
+# Written/run on - R Studio Desktop 
+# Version - 3.5.1
+#
+# install.packages("magrittr")
+# install.packages("tidyr")
+# install.packages("dplyr")
+# install.packages("tidylog")
+# install.packages("readr")
+# install.packages("glue")
+# install.packages("xfun")
+#
+# Description - Code for updating small area population estimates in the 
+#               Populations Datamart.
+# Approximate run time - 30 minutes
+
+library(magrittr)
+library(tidyr)
+library(dplyr)
+library(tidylog)
+library(readr)
+library(glue)
+library(xfun)
+
+# Set filepaths
+
+base_filepath <- file.path("//Freddy", "DEPT", "PHIBCS", "PHI", 
+                           "Referencing & Standards", "GPD", "2_Population")
+lookups_filepath <- file.path(base_filepath, "Small Area Population estimates", 
+                              "Lookup Files", "R Files")
+templates_filepath <- file.path(base_filepath, "Population Datamart", 
+                                "Creation of Files", "Templates", "R Templates")
+datamart_filepath <- file.path(base_filepath, "Population Datamart", 
+                               "Lookup Files", "Other Geographies")
+
+# Set date for filenames
+
+date <- strftime(Sys.Date(), format = "%Y%m%d")
+
+# Get CHP2012 column for matching
+
+DZ_CHP <- read_csv(paste0("//Isdsf00d03/cl-out/lookups/Unicode/Geography/", 
+                          "DataZone2011/DataZone2011.csv")) %>% 
+  select(DataZone2011, CHP2012) %>% 
+  rename(Data_Zone = DataZone2011, 
+         CHP_Code = CHP2012)
+  
+
+
+### 2 - Create Function for Outputs ----
+
+datamart_output <- function(start, end, pop_name, file, file_name, template){
+  
+  for (i in start:end){
+    
+    # Get most recent population estimates file
+    # Filter for required year
+    # Gather data into correct format
+    # Remove the age prefix from all age names
+    # Create Population_Name column and rename columns
+    # Match on CHP2012
+    # Create blank columns to fit datamart structure
+    # Reorder columns
+    
+    data <- readRDS(glue("{lookups_filepath}/{file}")) %>% 
+      filter(year == i) %>% 
+      gather(Age_Band, Population, "age0":"age90plus") %>% 
```

Comment:
We can also use pivot_longer function in the future. It replaces gather but only available in more recent tidyr.